gm4_soul_fragments/data/gm4_soul_fragments/functions/soul_fragment/process_check.mcfunction

```diff
@@ -0,0 +1,16 @@
+# @s = soul fragment
+# at @s
+# from main
+
+# check if there are items or players nearby before further processing
+execute if entity @a[distance=..17] run function gm4_soul_fragments:soul_fragment/process
+
+# check if there is soul fire (of any kind) nearby
+execute store result score @s gm4_sf_fire run clone ~-2 ~-1 ~-2 ~2 ~2 ~2 ~-2 ~-1 ~-2 filtered #gm4_soul_fragments:sf_fire force
+execute if score @s gm4_sf_fire matches 0 run execute store result score @s gm4_sf_fire run clone ~-2 ~-1 ~-2 ~2 ~2 ~2 ~-2 ~-1 ~-2 filtered #gm4_soul_fragments:sf_campfire[lit=true] force
```

Comment:
the block tag `#gm4_soul_fragments:sf_campfire` only has one block. It's not an issue runtime wise, but it's kinda odd unless you plan to add another block that also has the [lit=true] block state.